imod/mf6/pkgbase.py

```diff
@@ -1,16 +1,51 @@
 import abc
+import inspect
 import numbers
 import pathlib
+from functools import wraps
+from typing import Callable
 
 import numpy as np
 import xarray as xr
 import xugrid as xu
 
 import imod
 from imod.mf6.interfaces.ipackagebase import IPackageBase
+from imod.typing.grid import GridDataset, merge_with_dictionary
 
 TRANSPORT_PACKAGES = ("adv", "dsp", "ssm", "mst", "ist", "src")
 EXCHANGE_PACKAGES = "gwfgwf"
+ARGS_TO_EXCLUDE = ["validate"]
+
+
+def pkg_init(exclude_in_dataset: list[str]):
+    # Nested decorator to support arguments, see:
+    # https://stackoverflow.com/a/42581103
+    def pkg_init_decorator(init: Callable):
+        @wraps(init)
+        def merge_init(self, *args, **kwargs):
+            # Collect user specified args and kwargs
+            args_dict = (
+                inspect.signature(self.__init__).bind_partial(*args, **kwargs).arguments
+            )
+            # Add default values
+            for k, v in inspect.signature(self.__init__).parameters.items():
```

Comment:
you can improve clarity in this function 

- by adding comments. What are you looping over ( the arguments of the constructor being wrapped?)

- k, v = key , value? if so, write it out for clarity. What are their types? (since below we call v.default)